lib/elixir/lib/macro.ex

```diff
@@ -2280,4 +2280,178 @@ defmodule Macro do
   defp trim_leading_while_valid_identifier(other) do
     other
   end
+
+  # Made public since it's used by the Kernel.dbg/2 macro.
+  @doc false
+  def __default_dbg_fun__(ast, opts, env) do
+    default_opts = [
+      width: 80,
+      pretty: true
+    ]
+
+    opts = quote do: Keyword.merge(unquote(default_opts), unquote(opts))
+
+    opts =
+      quote do
+        if IO.ANSI.enabled?() do
+          Keyword.put_new(unquote(opts), :syntax_colors, unquote(dbg_default_syntax_colors()))
+        else
+          unquote(opts)
+        end
+      end
+
+    quote do
+      {formatted, result} = unquote(__default_dbg_fun_format__(ast, opts, env))
+      :ok = IO.write(formatted)
+      result
```

Comment:
We most likely want to reduce the amount of generated code.

Ideally, this would be something like this:

```elixir
stack_from_env = Exception.format_stacktrace(Macro.Env.stacktrace(env))

quote do
  result = unquote(modified_ast)
  Macro.__dbg__(result, unquote(code_to_string), unquote(stacktrace), unquote(opts))
end
```

Where `modified_ast` is at runtime something like `{:value, code, value}`, `{:pipe, [{first_pipe, result}, {second_pipe, result}, ...]}`.

But the other things, such as labels, printing, stacktrace, etc, can all be computed inside `__dbg__`.